From 11e58efba3f904a12b87c01cc55eed984140e662 Mon Sep 17 00:00:00 2001
From: Tomas Novotny <tomas.novotny@tbs-biometrics.com>
Date: Mon, 21 Jul 2014 12:53:24 +0200
Subject: [PATCH] sunxi: bee: Adjust config for our board

Signed-off-by: Tomas Novotny <novotny@rehivetech.com>
---
 include/configs/sunxi-common.h | 31 +++++++------------------------
 1 file changed, 7 insertions(+), 24 deletions(-)

diff --git a/include/configs/sunxi-common.h b/include/configs/sunxi-common.h
index 019a1bb..b42bdce 100644
--- a/include/configs/sunxi-common.h
+++ b/include/configs/sunxi-common.h
@@ -165,7 +165,7 @@
 	"console=ttyS0,115200\0" \
 	"panicarg=panic=10\0" \
 	"extraargs=\0" \
-	"loglevel=8\0" \
+	"loglevel=7\0" \
 	"scriptaddr=0x44000000\0" \
 	"device=mmc\0" \
 	"partition=0:1\0" \
@@ -180,7 +180,8 @@
 	  " setenv bootargs console=${console} root=${root}" \
 	  " loglevel=${loglevel} ${panicarg} ${extraargs}" \
 	  "\0" \
-	"kernel=uImage\0" \
+	"kernel=zImage\0" \
+	"dtb=sun4i-a10-olinuxino-lime.dtb\0" \
 	"bootenv=uEnv.txt\0" \
 	"bootscr=boot.scr\0" \
 	"script=script.bin\0" \
@@ -199,27 +200,8 @@
 	  "ext2load $device $partition $scriptaddr ${bootenv}" \
 	  "\0" \
 	"loadkernel=" \
-	  "if "\
-	    "bootpath=/boot/" \
-	    " && " \
-	    "ext2load $device $partition 0x43000000 ${bootpath}${script}" \
-	    " && " \
-	    "ext2load $device $partition 0x48000000 ${bootpath}${kernel}" \
-	  ";then true; elif " \
-	    "bootpath=/" \
-	    " && " \
-	    "fatload $device $partition 0x43000000 ${script}" \
-	    " && " \
-	    "fatload $device $partition 0x48000000 ${kernel}" \
-	  ";then true; elif " \
-	    "bootpath=/" \
-	    " && " \
-	    "ext2load $device $partition 0x43000000 ${bootpath}${script}" \
-	    " && " \
-	    "ext2load $device $partition 0x48000000 ${bootpath}${kernel}" \
-	  ";then true; else "\
-	    "false" \
-	  ";fi" \
+	  "fatload $device $partition 0x46000000 $kernel && " \
+	  "fatload $device $partition 0x49000000 $dtb" \
 	  "\0" \
 	"autoboot=" \
 	  "run loadkernel" \
@@ -228,7 +210,7 @@
 	  " && " \
 	  RESET_WATCHDOG \
 	  " && " \
-	  "bootm 0x48000000" \
+	  "bootz 0x46000000 - 0x49000000" \
 	  "\0" \
 	"boot_ram=" \
 	  "saved_stdout=$stdout;setenv stdout nc;"\
@@ -242,6 +224,7 @@
 	  "\0" \
 	""
 
+#define CONFIG_BOOTDELAY	2
 #define CONFIG_SYS_BOOT_GET_CMDLINE
 
 #include <config_cmd_default.h>
-- 
1.8.3.1

