From 4410dd8a713e36e5b185b652de8d8ffc0e0d2cad Mon Sep 17 00:00:00 2001
From: Tomas Novotny <novotny@rehivetech.com>
Date: Fri, 25 Oct 2013 14:52:49 +0200
Subject: [PATCH 1/2] sunxi: bee: Use lower debug level

Signed-off-by: Tomas Novotny <novotny@rehivetech.com>
---
 include/configs/sunxi-common.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/configs/sunxi-common.h b/include/configs/sunxi-common.h
index d8b3ee0..fe82803 100644
--- a/include/configs/sunxi-common.h
+++ b/include/configs/sunxi-common.h
@@ -191,7 +191,7 @@
 	"console=ttyS0,115200\0" \
 	"panicarg=panic=10\0" \
 	"extraargs=\0" \
-	"loglevel=8\0" \
+	"loglevel=5\0" \
 	"scriptaddr=0x44000000\0" \
 	"device=mmc\0" \
 	"partition=0:1\0" \
-- 
1.7.11.3

