From 869c7996964dd5d75699afee23b9352be483c506 Mon Sep 17 00:00:00 2001
From: Matej Postolka <xposto02@stud.fit.vutbr.cz>
Date: Mon, 29 Jun 2015 20:10:59 +0200
Subject: [PATCH 6/9] Modified boot sequence execution based on SRAM evaluation

Signed-off-by: Matej Postolka <xposto02@stud.fit.vutbr.cz>
---
 include/config_distro_bootcmd.h | 44 +++++++++++++++++++++++++++++++++++++++++
 1 file changed, 44 insertions(+)

diff --git a/include/config_distro_bootcmd.h b/include/config_distro_bootcmd.h
index 540a8f6..afa4194 100644
--- a/include/config_distro_bootcmd.h
+++ b/include/config_distro_bootcmd.h
@@ -30,6 +30,50 @@
 #define CONFIG_PARTITION_UUIDS
 #define CONFIG_CMD_PART
 
+#define BOOTENV_SHARED_BLKDEV_BODY_MMC(devtypel) \
+		"if " #devtypel " dev ${devnum}; then " \
+			"setenv devtype " #devtypel "; " \
+			"setenv mmc_devnum 0; " \
+			"run scan_dev_for_boot; " \
+			"run read_sram; " \
+			"if mcheck $bootcounter_addr_r; then " \
+			"echo SRAM OK; else " \
+			"echo SRAM IS CORRUPT; " \
+			"run initialize_sram; fi; " \
+			"if bootcheck $bootcounter_addr_r; then " \
+			"run mmc_boot_zimage_recovery; else " \
+			"run mmc_boot_zimage; fi; " \
+		"fi\0"
+
+#define BOOTENV_SHARED_BLKDEV_MMC(devtypel) \
+	#devtypel "_boot=" \
+	BOOTENV_SHARED_BLKDEV_BODY_MMC(devtypel)
+
+#define BOOTENV_LEGACY_ZIMAGE \
+	"i2c_busnum=2\0" \
+	"i2c_sram_addr=0x6f\0" \
+	"i2c_sram_offset=0x20\0" \
+	"i2c_sram_count=0x40\0" \
+	\
+	"mmc_boot_zimage=" \
+		"echo Booting zImage - REGULAR MODE...; " 		\
+		"run increment_sram; "				\
+		"fatload mmc ${mmc_devnum} ${kernel_addr_r} zImage; " 	\
+		"fatload mmc ${mmc_devnum} ${fdt_addr_r} ${fdtfile}; "	\
+		"setenv bootargs console=${console} "			\
+		"root=/dev/mmcblk0p2 rw rootwait panic=12; "		\
+		"bootz ${kernel_addr_r} - ${fdt_addr_r}\0"		\
+		\
+	"mmc_boot_zimage_recovery=" \
+		"echo Booting zImage - RECOVERY MODE...; "		\
+		"run increment_sram; "				\
+		"fatload mmc ${mmc_devnum} ${kernel_addr_r} zImage_recovery; " 	\
+		"fatload mmc ${mmc_devnum} ${fdt_addr_r} ${fdtfile}; "	\
+		"setenv bootargs console=${console} "			\
+		"root=/dev/mmcblk0p3 ro rootwait panic=12; "		\
+		"bootz ${kernel_addr_r} - ${fdt_addr_r}\0"
+
+
 #define BOOTENV_SHARED_BLKDEV_BODY(devtypel) \
 		"if " #devtypel " dev ${devnum}; then " \
 			"setenv devtype " #devtypel "; " \
-- 
1.9.1

