--- a/Makefile
+++ b/Makefile
@@ -6,7 +6,7 @@
 VERSN=$(VERS).$(shell svn info | grep 'Revision' | awk '{ print $$2; }')
 
 RELATIVE_PATH=YES
-BINDIR=/opt/fhem
+BINDIR=/usr/share/fhem
 MODDIR=$(BINDIR)
 VARDIR=$(BINDIR)/log
 MANDIR=$(BINDIR)/docs
@@ -57,13 +57,13 @@
 	@rm fhem.cfg.install
 	@cp fhem.pl $(RBINDIR)
 	@cp -rp FHEM docs www contrib configDB.pm $(RMODDIR)
-	@cp -rp README_DEMO.txt demolog fhem.cfg.demo $(RDEMODIR)
 	@cp docs/fhem.man $(RMANDIR)/fhem.pl.1
 	@gzip -f -9 $(RMANDIR)/fhem.pl.1
 	@echo "- cleanup: removing .svn leftovers"
 	@find $(RMODDIR) -name .svn -print | xargs rm -rf
 	@find $(RMODDIR) -name svn-commit\* -print | xargs rm -rf
-	@echo
+	@echo "Remove lcd4linux (due to architecture-dependent binaries)"
+	@rm -rf $(RMODDIR)/contrib/lcd4linux
 	@echo "Installation of fhem completed!"
 	@echo
 	@echo "Start fhem with"
