From 63d7058a032097babdb4ce2ce1eeb22dfe3b8c6b Mon Sep 17 00:00:00 2001
From: Tomas Novotny <tomas@novotny.cz>
Date: Thu, 11 Feb 2016 14:13:08 +0100
Subject: [PATCH] ini: configure for production use

Signed-off-by: Tomas Novotny <tomas@novotny.cz>
---
 etc/AdaApp.ini | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/etc/AdaApp.ini b/etc/AdaApp.ini
index acbbc4d..9666f86 100644
--- a/etc/AdaApp.ini
+++ b/etc/AdaApp.ini
@@ -1,12 +1,10 @@
 [Adapter]
-debug = true
-id = 0x11
+debug = false
 
 ; Server connection
 [Server]
-;port = 9092
-port = 7080
-ip = 147.229.176.131
+port = 9092
+ip = 37.157.199.235
 
 [Versions]
 protocol = 1.0
-- 
1.8.3.1

