From 0d82b04e439cceec903d919f0ad65ed299f5571f Mon Sep 17 00:00:00 2001
From: Tomas Novotny <tomas@novotny.cz>
Date: Mon, 14 Dec 2015 15:40:22 +0100
Subject: [PATCH] ini: configure for production use

Signed-off-by: Tomas Novotny <tomas@novotny.cz>
---
 etc/AdaApp.ini | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/etc/AdaApp.ini b/etc/AdaApp.ini
index 9368b0b..431cf2e 100644
--- a/etc/AdaApp.ini
+++ b/etc/AdaApp.ini
@@ -1,12 +1,10 @@
 [Adapter]
-debug = true
-id = 0x11
+debug = false
 
 ; Server connection
 [Server]
-;port = 9092
-port = 7080
-ip = 147.229.176.131
+port = 9092
+ip = 37.157.199.235
 listen_port = 7978
 
 [Versions]
-- 
1.8.3.1

