From fc48cc8f46a837b258ad68eeff89cd7a63ae1f59 Mon Sep 17 00:00:00 2001
From: Matej Postolka <xposto02@stud.fit.vutbr.cz>
Date: Mon, 29 Jun 2015 20:12:23 +0200
Subject: [PATCH 07/11] Lowered boot delay to 1 second

Signed-off-by: Matej Postolka <xposto02@stud.fit.vutbr.cz>
---
 include/config_distro_defaults.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/config_distro_defaults.h b/include/config_distro_defaults.h
index bd8fbca..83cfc46 100644
--- a/include/config_distro_defaults.h
+++ b/include/config_distro_defaults.h
@@ -54,7 +54,7 @@
 
 #define CONFIG_CMDLINE_EDITING
 #define CONFIG_AUTO_COMPLETE
-#define CONFIG_BOOTDELAY     2
+#define CONFIG_BOOTDELAY     1
 #define CONFIG_SYS_LONGHELP
 #define CONFIG_MENU
 #define CONFIG_DOS_PARTITION
-- 
1.9.1

